components/esp_modem/src/esp_modem_dce.cpp

```diff
@@ -21,10 +21,17 @@ static bool exit_data(DTE &dte, ModuleIf &device, Netif &netif)
     netif.stop();
     auto signal = std::make_shared<SignalGroup>();
     std::weak_ptr<SignalGroup> weak_signal = signal;
-    dte.set_read_cb([weak_signal](uint8_t *data, size_t len) -> bool {
+    dte.set_read_cb([&netif, weak_signal](uint8_t *data, size_t len) -> bool {
```

Comment:
I think we already had this discussion, but, is it safe to capture `netif` by reference here?

Reply:
Yes, it is safe (in the current design), as DCE has exclusive ownership of the netif class.
Passing this to the callback should be safe too, as the callback is valid in the current scope and we get the reference from the function params.